otj-metrics-core/src/main/java/com/opentable/metrics/ready/Result.java

```diff
@@ -0,0 +1,168 @@
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The result of a {@link ReadyCheck} being run. It can be ready (with an optional message and optional details)
+ * or unready (with either an error message or a thrown exception and optional details).
+ */
+public class Result {
+    private static final int PRIME = 31;
+
+    /**
+     * Returns a ready {@link Result} with no additional message.
+     *
+     * @return a ready {@link Result} with no additional message
+     */
+    public static Result ready() {
+        return new Result(true, null, null);
+    }
+
+    /**
+     * Returns a ready {@link Result} with an additional message.
+     *
+     * @param message an informative message
+     * @return a ready {@link Result} with an additional message
+     */
+    public static Result ready(String message) {
+        return new Result(true, message, null);
+    }
+
+    /**
+     * Returns a ready {@link Result} with a formatted message.
+     * <p/>
+     * Message formatting follows the same rules as {@link String#format(String, Object...)}.
+     *
+     * @param message a message format
+     * @param args    the arguments apply to the message format
+     * @return a ready {@link Result} with an additional message
+     * @see String#format(String, Object...)
+     */
+    public static Result ready(String message, Object... args) {
+        return ready(String.format(message, args));
+    }
+
+    /**
+     * Returns an unready {@link Result} with the given message.
+     *
+     * @param message an informative message describing how the ready check failed
+     * @return an unready {@link Result} with the given message
+     */
+    public static Result unready(String message) {
+        return new Result(false, message, null);
+    }
+
+    /**
+     * Returns an unready {@link Result} with a formatted message.
+     * <p/>
+     * Message formatting follows the same rules as {@link String#format(String, Object...)}.
+     *
+     * @param message a message format
+     * @param args    the arguments apply to the message format
+     * @return an unready {@link Result} with an additional message
+     * @see String#format(String, Object...)
+     */
+    public static Result unready(String message, Object... args) {
+        return unready(String.format(message, args));
+    }
+
+    /**
+     * Returns an unready {@link Result} with the given error.
+     *
+     * @param error an exception thrown during the ready check
+     * @return an unready {@link Result} with the given {@code error}
+     */
+    public static Result unready(Throwable error) {
+        return new Result(false, error.getMessage(), error);
+    }
+
+    private final boolean ready;
+    private final String message;
+    private final Throwable error;
+    private final String timestamp =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXXX").format(ZonedDateTime.now());
```

Comment:
Do we need this level of pression? We could just use the most precise standard ISO formatter that's built-in

Reply:
It's just a copy/paste. It's conssitent with HealthCheck whhich we can't chhange